app/src/main/java/com/hamdan/forzenbook/view/login/MainLogin.kt

```diff
@@ -88,119 +90,115 @@ private fun Content(
     val focusManager = LocalFocusManager.current
     val keyboardController = LocalSoftwareKeyboardController.current
 
-    if (isInputtingCode) {
-        InputField(
-            label = stringResource(R.string.login_email_prompt), value = email,
-            onValueChange = {
-                email = it
-                emailError = email.length > EMAIL_LENGTH_LIMIT || !validateEmail(email)
-                onTextChange(Entry(email, emailError), Entry(code, codeError), isInputtingCode)
-            },
-            imeAction = ImeAction.Next, onNext = { focusManager.moveFocus(FocusDirection.Down) }
-        )
-        if (emailError && email.isNotEmpty()) {
-            Text(text = stringResource(R.string.login_email_error))
+    InputField(
+        label = stringResource(R.string.login_email_prompt), value = email,
+        onValueChange = {
+            email = it
+            emailError = when (true) {
+                (email.length > EMAIL_LENGTH_LIMIT) -> LoginError.EmailError.Length
+                (!TextUtils.isEmpty(email) && !validateEmail(email)) -> LoginError.EmailError.InvalidFormat
+                else -> LoginError.EmailError.Valid
+            }
+            onTextChange(Entry(email, emailError), Entry(code, codeError), isInputtingCode)
+        },
+        imeAction = if (isInputtingCode) {
+            ImeAction.Next
+        } else {
+            ImeAction.Done
+        }, onNext = if (!isInputtingCode) {
+            { focusManager.moveFocus(FocusDirection.Down) }
+        } else {
+            {}
```

Comment:
Why just an empty bracket set here? Maybe making it `null` would look cleaner if it accepts that?

Reply:
no it doesn't accept null sadly, it wants a non null lamda

Reply:
I did realize theres no real reason to give it an empty lambda though, as the action won't be called if its not the current step, so no need for the weird if else